iodata/formats/qchemlog.py

```diff
@@ -0,0 +1,368 @@
+# IODATA is an input and output module for quantum chemistry.
+# Copyright (C) 2011-2019 The IODATA Development Team
+#
+# This file is part of IODATA.
+#
+# IODATA is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# IODATA is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+# --
+"""Q-Chem Log file format.
+
+This module will load Q-Chem log file into IODATA.
+"""
+
+
+import re
+from typing import Tuple
+
+import numpy as np
+
+from ..docstrings import document_load_one
+from ..orbitals import MolecularOrbitals
+from ..periodic import sym2num
+from ..utils import LineIterator, kcalmol, calmol
+
+__all__ = []
+
+PATTERNS = ['*.qchemlog']
+
+
+@document_load_one("qchemlog", ['atcoords', 'atmasses', 'atnums', 'charge', 'energy', 'g_rot',
+                                'mo', 'lot', 'nelec', 'obasis_name', 'run_type', 'extra'],
+                   ['athessian'])
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    data = load_qchemlog_low(lit)
+
+    # add these labels if they are loaded
+    result_labels = ['atcoords', 'atmasses', 'atnums', 'charge', 'charge', 'energy', 'g_rot',
+                     'run_type', 'athessian', 'lot', 'obasis_name']
+    result = {label: data[label] for label in result_labels if data.get(label) is not None}
+    # add number of electrons
+    result['nelec'] = data['alpha_elec'] + data['beta_elec']
+
+    # mulliken charges
+    if data.get("mulliken_charges") is not None:
+        result["atcharges"] = {"mulliken": data["mulliken_charges"]}
+
+    # build molecular orbitals
+    # ------------------------
+    # restricted case
+    if not data['unrestricted']:
+        mo_energies = np.concatenate((data['mo_a_occ'], data['mo_a_vir']), axis=0)
+        mo_coeffs = np.empty((data['nbasis'], data['norba'])) * np.nan
+        mo_occs = np.zeros(mo_coeffs.shape[1])
+        mo_occs[:data['alpha_elec']] = 1.0
+        mo_occs[:data['beta_elec']] += 1.0
+        mo = MolecularOrbitals("restricted", data['norba'], data['norba'],
+                               mo_occs, mo_coeffs, mo_energies, None)
+    # unrestricted case
+    else:
+        mo_energies = np.concatenate((data['mo_a_occ'], data['mo_a_vir'],
+                                      data['mo_b_occ'], data['mo_b_vir']), axis=0)
+        mo_coeffs = np.empty((data['nbasis'], data['norba'] + data['norbb'])) * np.nan
```

Comment:
same comment as above.

Reply:
Fixed in 45eeba0. Thanks for pointing it out.